core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -352,7 +352,7 @@ public void commit() {
             });
     LOG.info("Committed snapshot changes");
 
-    if (cleanExpiredFiles) {
+    if (cleanExpiredFiles && !base.snapshots().isEmpty()) {
```

Comment:
This seems odd. Shouldn't this happen automatically when `cleanExpiredSnapshots()` gathers the set of snapshots to clean?